Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -182,21 +182,37 @@ + (NSInteger)getRetryCount {
     return retryCount;
 }
 
-+ (NSString *)getAppKey {
-    NSString *ret = [[[NSBundle mainBundle] infoDictionary] objectForKey:KEY_APP_KEY];
-    if (!ret || ret.length == 0) {
-        // for backward compatibility
-        ret = [BNCPreferenceHelper readStringFromDefaults:KEY_APP_KEY];
-        if (!ret) {
-            ret = NO_STRING_VALUE;
++ (NSString *)getBranchKey {
+    if (!Branch_Key) {
+        Branch_Key = [BNCPreferenceHelper getBranchKey:YES];
+    }
+    
+    return Branch_Key;
+}
+
++ (NSString *)getBranchKey:(BOOL)isLive {
+    NSString *key = nil;
+    
+    id ret = [[[NSBundle mainBundle] infoDictionary] objectForKey:KEY_BRANCH_KEY];
+    if (ret) {
+        if ([ret isKindOfClass:[NSString class]]) {
+            key = ret;
+        } else if ([ret isKindOfClass:[NSDictionary class]]) {
```

Comment:
Is there ever a case where this could be saved as a dictionary? I don't see any code here that inserts it in that manner.